src/models/quote.js

```diff
@@ -120,23 +146,50 @@ function * getAccountLedger (account) {
   return ledger
 }
 
-function * getQuote (params, ledgers, config) {
-  const query = yield makeQuoteQuery(params, config)
-  const quote = yield backend.getQuote(makeQuoteArgs(query))
+function validatePrecision (amount, precision, ledger) {
+  if (new BigNumber(amount).precision() > precision) {
+    throw new UnacceptableAmountError(`Amount (${amount}) exceeds ledger precision on ${ledger} ledger`)
+  }
+}
 
+function * validateBalance (query, quote) {
   const sourceBalance = yield balanceCache.get(query.source_ledger)
   const sourceAmount = new BigNumber(quote.source_amount)
   if (sourceBalance.lessThan(sourceAmount)) {
     throw new UnacceptableAmountError('Insufficient liquidity in market maker account')
   }
+}
+
+function * getQuote (params, ledgers, config) {
+  const query = yield makeQuoteQuery(params, config)
+  const quote = yield backend.getQuote((yield makeQuoteArgs(query)))
+
+  const sourcePrecisionAndScale = yield getPrecisionAndScale(query.source_ledger)
+  const dstPrecisionAndScale = yield getPrecisionAndScale(query.destination_ledger)
+
+  const roundedSourceAmount = new BigNumber(quote.source_amount).toFixed(
+    sourcePrecisionAndScale.scale, BigNumber.ROUND_UP)
+
+  const roundedDestinationAmount = new BigNumber(quote.destination_amount).toFixed(
+    dstPrecisionAndScale.scale, BigNumber.ROUND_DOWN)
+
+  validatePrecision(roundedSourceAmount, sourcePrecisionAndScale.precision, 'source')
+  validatePrecision(roundedDestinationAmount, dstPrecisionAndScale.precision, 'destination')
+
+  const roundedQuote = _.assign(_.clone(quote), {
+    source_amount: roundedSourceAmount,
+    destination_amount: roundedDestinationAmount
+  })
```

Comment:
What do you think the behavior should be if either the source or destination amounts are rounded down to zero? Should it return a quote with an amount set to 0? Should it ensure that both numbers are at least the minimum allowed by the ledger?